packages/serverpod/lib/src/server/websocket_request_handlers/method_websocket_request_handler.dart

```diff
@@ -210,13 +210,24 @@ class _MethodStreamManager {
     required Server server,
     required WebSocket webSocket,
   }) {
-    if (endpointMethodConnector is! MethodConnector) {
-      // TODO: This is a temporary solution and is fixed in later commits.
-      throw Exception('MethodConnector is not a MethodConnector');
-    }
-
     var controller = StreamController<String>();
-    _handleStream(endpointMethodConnector, session, args, message, server);
+    if (endpointMethodConnector is MethodStreamConnector) {
+      _handleMethodStreamEndpoint(
+        endpointMethodConnector,
+        session,
+        args,
+        message,
+        server,
+      );
+    } else if (endpointMethodConnector is MethodConnector) {
+      _handleMethodCallEndpoint(
+        endpointMethodConnector,
+        session,
+        args,
+        message,
+        server,
+      );
+    }
```

Comment:
Discussion: To drive home the point, here we have separated the different methods! Instead of a bool toggle.

Reply:
Just for clarity, this is a temporary solution, MethodConnector will not be accepted in this context in the future. It will give the same result as having failed the method lookup.